src/text-filter/__tests__/text-filter.test.tsx

```diff
@@ -137,6 +140,52 @@ describe('countText', () => {
     expect(document.getElementById(ariaDescribedby[0])).toHaveTextContent('10 matches');
     expect(ariaDescribedby[1]).toBe('test-description');
   });
+
+  describe('live announcement', () => {
+    function getPoliteRegion() {
+      return document.querySelector('[aria-live=polite]')!;
+    }
+
+    beforeAll(() => {
+      jest.useFakeTimers();
+    });
+
+    afterAll(() => {
+      jest.useRealTimers();
+    });
+
+    test('includes the live announcement when all conditions met', () => {
+      renderTextFilter(<TextFilter filteringText="test" countText="10 matches" />);
+      jest.runAllTimers();
+      expect(getPoliteRegion()).toHaveTextContent('10 matches');
+    });
+
+    test('does not include the live announcement when loading = true', () => {
+      renderTextFilter(<TextFilter filteringText="test" loading={true} countText="10 matches" />);
+      jest.runAllTimers();
+      expect(getPoliteRegion()).toBeNull();
+    });
+
+    test('includes the live announcement when loading switches from true to false', () => {
+      const { rerender } = renderTextFilter(<TextFilter filteringText="test" loading={true} countText="10 matches" />);
+      jest.runAllTimers();
+      expect(getPoliteRegion()).toBeNull();
+
+      rerender(<TextFilter filteringText="test" loading={false} countText="123 matches" />);
```

Comment:
In this test, the `countText` also changes from `10 matches` to `123 matches`. I suggest to use the same countText for both renders, so that we can be more certain that the live announcement is caused by the loading and not by the countText changing (we test that in the test below anyway)